crates/cxx-qt-gen/src/parser/qobject.rs

```diff
@@ -66,31 +65,29 @@ pub struct ParsedQObject {
 }
 
 impl ParsedQObject {
-    /// Parse a ForeignTypeIdentAlias into a [ParsedQObject] with the index of the cxx_qt::qobject specified
-    pub fn from_foreign_item_type(
-        qobject_ty: &ForeignTypeIdentAlias,
-        attr_index: usize,
-    ) -> Result<Self> {
-        let qml_metadata = Self::parse_qml_metadata(qobject_ty, attr_index)?;
-
-        let attrs_map = attribute_tokens_to_map::<Ident, LitStr>(
-            &qobject_ty.attrs[attr_index],
-            AttributeDefault::Some(|span| LitStr::new("", span)),
-        )?;
+    /// Parse a ForeignTypeIdentAlias into a [ParsedQObject] with the index of the #[qobject] specified
+    pub fn from_foreign_item_type(qobject_ty: &ForeignTypeIdentAlias) -> Result<Self> {
+        let mut qobject_ty = qobject_ty.clone();
+
+        // Find any QML metadata
+        let qml_metadata = Self::parse_qml_metadata(&qobject_ty.ident_left, &mut qobject_ty.attrs)?;
 
         // Find if there is any base class
-        let base_class = attrs_map
-            .get(&quote::format_ident!("base"))
-            .map(|base| base.value());
+        let base_class = if let Some(index) = attribute_find_path(&qobject_ty.attrs, &["base"]) {
+            let attr = qobject_ty.attrs.remove(index);
+            Some(expr_to_string(&attr.meta.require_name_value()?.value)?)
+        } else {
+            None
+        };
 
         // Load the namespace, if it is empty then the ParsedCxxQtData will inject any global namespace
-        let namespace = attrs_map
-            .get(&quote::format_ident!("namespace"))
-            .map_or_else(|| "".to_owned(), |base| base.value());
-
-        // Remove the macro from the type
-        let mut qobject_ty = qobject_ty.clone();
-        qobject_ty.attrs.remove(attr_index);
+        let namespace = if let Some(index) = attribute_find_path(&qobject_ty.attrs, &["namespace"])
```

Comment:
It probably makes sense to introduce a `attribute_take_path` function.
e.g.:
``` rust
pub fn attribute_find_path(attrs: &mut Vec<Attribute>, path: &[&str]) -> Option<Attribute> { ... } 
```
Which would directly take the attribute out if it's found.
That would reduce duplicate code 4 times just in this file.

Reply:
i'll add another commit with this

Reply:
done